File: 3262-find-polygon-with-the-largest-perimeter/find-polygon-with-the-largest-perimeter.java
class Solution {
    public long largestPerimeter(int[] nums) {
        Arrays.sort(nums);
        long  maximumPerimeter = Integer.MIN_VALUE;
        long prefixSum = nums[0] + nums[1];

        for (int i = 2; i < nums.length; i++) {
            if (prefixSum > (long)nums[i]) {
                maximumPerimeter = Math.max(maximumPerimeter, prefixSum + nums[i]);
            }
            prefixSum += nums[i];
        }

        return (maximumPerimeter == Integer.MIN_VALUE) ? -1 : maximumPerimeter;
    }
}

File: 3262-find-polygon-with-the-largest-perimeter/find-polygon-with-the-largest-perimeter.cpp
class Solution {
public:
    long long largestPerimeter(vector<int>& nums) {
        sort(nums.begin(), nums.end());
        long long maximumPerimeter = INT_MIN;
        long long prefixSum = nums[0] + nums[1];

        for (long long i = 2; i < nums.size(); i++) {
            if (prefixSum > nums[i]) {
                maximumPerimeter = max(maximumPerimeter, prefixSum + nums[i]);
            }
            prefixSum += nums[i];
        }

        return (maximumPerimeter == INT_MIN) ? -1 : maximumPerimeter;
    }
};